src/Microsoft.Health.ControlPlane.Core/Features/Rbac/IdentityProvider.cs

```diff
@@ -40,5 +43,68 @@ public IdentityProvider(string name, string authority, IReadOnlyList<string> aud
 
         [JsonProperty("version")]
         public virtual string Version { get; protected set; }
+
+        public virtual IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
+        {
+            if (string.IsNullOrWhiteSpace(Name))
+            {
+                yield return new ValidationResult(Resources.IdentityProviderNameEmpty);
+            }
+
+            if (string.IsNullOrWhiteSpace(Authority))
+            {
+                yield return new ValidationResult(Resources.IdentityProviderAuthorityEmpty);
+            }
+
+            if (Audience == null)
+            {
+                yield return new ValidationResult(Resources.IdentityProviderAudienceIsNull);
+            }
+
+            foreach (var audience in Audience)
+            {
+                if (string.IsNullOrWhiteSpace(audience))
+                {
+                    yield return new ValidationResult(Resources.IdentityProviderInvalidAudience);
+                    break;
+                }
+            }
+
+            foreach (ValidationResult result in ValidateAuthority())
+            {
+                yield return result;
+            }
+        }
+
+        public virtual IEnumerable<ValidationResult> ValidateAuthority()
+        {
+            using (var handler = new HttpClientHandler())
```

Comment:
This may not work with the E2E tests with identity server. It won't be available through the standard http handler and has a special one use.

Reply:
Brandon and I spoke. We will have to test this when we write integration tests. At that point if there is an issue, we can register one using Dependency Injection.